dev/service-registry/client-core/src/test/scala/com/lightbend/lagom/internal/registry/AbstractLoggingServiceRegistryClientSpec.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.lightbend.lagom.internal.registry
+
+import java.net.URI
+
+import org.scalatest.{ AsyncWordSpec, Matchers }
+
+import scala.concurrent.Future
+
+class AbstractLoggingServiceRegistryClientSpec extends AsyncWordSpec with Matchers {
+
+  private val client = new AbstractLoggingServiceRegistryClient {
+    override def internalLocateAll(serviceName: String): Future[List[URI]] = serviceName match {
+      case "failing-service"    => Future.failed(new IllegalArgumentException("Ignore: expected error"))
+      case "empty-service"      => Future.successful(List())
+      case "successful-service" => Future.successful(List(URI.create("http://localhost:8080")))
+    }
+  }
+
+  "AbstractLoggingServiceRegistryClient" when {
+    /*
+     * This class is very simple, and the tests are straightforward, but it's
+     * still useful to run the tests to manually inspect log messages.
+     */
```

Comment:
Not sure it is important, but, on the Akka side, we often use the `EventFilter` from the testkit for this kind of thing, to actually assert the log entries happened. Here's an example test case where we do that: https://github.com/akka/akka/blob/8c4181e5526c42e739553bff953f77c63b8260c0/akka-actor-typed-tests/src/test/scala/akka/actor/typed/WatchSpec.scala (typed, but the event filter is from the untyped testkit)

Reply:
I thought about doing that, but right now it has no dependency on the actor system and logs using slf4j.

We had a bit of a conversation at the team meeting in Belgium about whether we should standardize on Akka logging everywhere, slf4j logging everywhere, or continue the mix that we have now (for both Lagom and Play internals). It was inconclusive and the uneasy consensus was to keep the existing mix.

I'm willing to be persuaded one way or another. @ignasi35 / @marcospereira do you have an opinion?

Reply:
Ah, I thought Lagom was on top of Akka logging. Thx for clarifying.